addons/calendar/controllers/main.py

```diff
@@ -91,3 +92,20 @@ def notify(self):
     @http.route('/calendar/notify_ack', type='json', auth="user")
     def notify_ack(self):
         return request.env['res.partner'].sudo()._set_calendar_last_notif_ack()
+
+    @http.route('/calendar/ics', type='http', auth='basic')
+    def calendar_ics(self, *args, **kwargs):
+        if request.httprequest.headers.get('Accept', 'text/calendar') != 'text/calendar':
+            raise NotAcceptable()
+
+        data = self._ics()
+        headers = [
+            ('Content-Disposition', http.content_disposition(f"{http.request.env.user.login}.ics")),
+            ('Content-Length', len(data)),
+            ('Content-Type', 'text/calendar'),
+        ]
+        return http.request.make_response(data, headers)
+
+    def _ics(self):
+        domain = [('user_id', '=', http.request.env.user.id)]
+        return http.request.env['calendar.event'].search(domain).ics()
```

Comment:
I'm still reading up on iCalendar `LAST-MODIFIED`, but am hopeful that some combination of that property and an HTTP `Last-Modified` header could let me filter here by `write_date`. I have been testing with Apple Simulator and haven't seen any relevant headers or GET params yet.